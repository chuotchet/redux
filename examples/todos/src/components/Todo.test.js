import React from 'react';
import { shallow } from 'enzyme';

import Todo from './Todo';

const setup = (setupProps = {}) => {
    const defaultProps = {
        completed: false,
        onClick: jest.fn(),
        text: 'Test',
        createdTime: '2018-10-12 16:58:30'
    }
    const props = {...defaultProps,...setupProps};
    const wrapper = shallow(
        <Todo onClick={props.onClick} completed={props.completed} text={props.text}  createdTime={props.createdTime}/>);

    return {
        props,
        wrapper
    };
};

describe('Todo', () => {
    it('should match snapshot', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    })
    it('should handle click button', () => {
        const { props, wrapper } = setup();

        expect(wrapper.shallow().find('li').props('style').style.textDecoration).toEqual('none');
        const link = wrapper.shallow().find('li');
        link.simulate('click');

        expect(props.onClick).toBeCalled();
    })
    it('should toogle style', () => {
        const { props, wrapper } = setup({completed: true});

        expect(wrapper.shallow().find('li').props('style').style.textDecoration).toEqual('line-through');
    })
})