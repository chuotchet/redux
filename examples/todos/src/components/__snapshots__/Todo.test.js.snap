// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should match snapshot 1`] = `
Object {
  "props": Object {
    "completed": false,
    "createdTime": "2018-10-12 16:58:30",
    "onClick": [MockFunction],
    "text": "Test",
  },
  "wrapper": ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <Todo
      completed={false}
      createdTime="2018-10-12 16:58:30"
      onClick={[MockFunction]}
      text="Test"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Test",
          " - ",
          "2018-10-12 16:58:30",
        ],
        "onClick": [MockFunction],
        "style": Object {
          "textDecoration": "none",
        },
      },
      "ref": null,
      "rendered": Array [
        "Test",
        " - ",
        "2018-10-12 16:58:30",
      ],
      "type": "li",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Test",
            " - ",
            "2018-10-12 16:58:30",
          ],
          "onClick": [MockFunction],
          "style": Object {
            "textDecoration": "none",
          },
        },
        "ref": null,
        "rendered": Array [
          "Test",
          " - ",
          "2018-10-12 16:58:30",
        ],
        "type": "li",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
}
`;
